homeassistant/components/overkiz/water_heater_entities/domestic_hot_water_production.py

```diff
@@ -58,17 +55,21 @@ def __init__(
         super().__init__(device_url, coordinator)
 
         # Init operation mode to set for this specific device
-        self.overkiz_to_operation_mode: dict[str, str] = {}
+        self.operation_mode_to_overkiz: dict[str, str] = {}
+        self._attr_operation_list = []
         state_mode_definition = self.executor.select_definition_state(
             OverkizState.IO_DHW_MODE, OverkizState.MODBUSLINK_DHW_MODE
         )
-        if state_mode_definition and state_mode_definition.values:
+        for param, mode in OVERKIZ_TO_OPERATION_MODE.items():
             # Filter only for mode allowed by this device
-            for param, mode in OVERKIZ_TO_OPERATION_MODE.items():
-                if param in state_mode_definition.values:
-                    self.overkiz_to_operation_mode[param] = mode
-        else:
-            self.overkiz_to_operation_mode = OVERKIZ_TO_OPERATION_MODE
+            # or allow all if no mode definition found
+            if (
+                state_mode_definition
+                and state_mode_definition.values
+                and param in state_mode_definition.values
+            ) or (not state_mode_definition):
```

Comment:
I'd start with the not case and then follow with the or case. We only need a single parenthesis then. It's also faster and easier to read since we can short circuit the check.

```py
if (
    not state
    or state
    and state.attribute
    and ...
):
```